client/app/shared/ace-editor/ace-editor.directive.js

```diff
@@ -0,0 +1,43 @@
+/** @ngInject */
```

Comment:
This feels like it should be a component and not a directive. In Angular2 this would be a template simply because it has a template and controller.

The controller should also take advantage of component lifecycle hooks, `$onInit`, `$onChanges` or `$onDestroy, `$postLink`.